### PR TITLE
feat: add analytics report to CI

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
This will eventually allow us to see how often a formula is installed, to grasp how much effort is warranted to fix it.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -87,6 +87,41 @@ jobs:
         env:
           TESTING_FORMULAE: ${{ steps.formulae-detect.outputs.formulae_to_fetch }}
         run: brew test-bot --only-bottles-fetch --testing-formulae="$TESTING_FORMULAE"
+  report_analytics:
+    runs-on: ubuntu-latest
+    needs: formulae_detect
+    steps:
+
+      - name: Publish Analytics to Summary
```
Comment (carlocab) [inline .github/workflows/tests.yml:95]:
```suggestion

  report_analytics:
    runs-on: ubuntu-latest
    needs: formulae_detect
    steps:
      - name: Publish Analytics to Summary
```

**Review by carlocab — APPROVED** [review]
Still draft?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -88,6 +88,41 @@ jobs:
...
+            echo "### Analytics about: ${TESTING_FORMULAE}"
+            echo "<details><summary>Click to expand</summary>"
+            echo ""
+            echo "| Formula | Errors | 30d | 90d | 356d |"
+            echo "|---|---|---|---|---|"
+            for formula in ${TESTING_FORMULAE//,/ }
+            do
+                FORMULA_30="$(echo "$DATA_30" | jq -r ".formulae[\"$formula\"][0].count")"
+                FORMULA_90="$(echo "$DATA_90" | jq -r ".formulae[\"$formula\"][0].count")"
+                FORMULA_1="$(echo "$DATA_1" | jq -r ".formulae[\"$formula\"][0].count")"
```
Comment (ZhongRuoyu) [inline .github/workflows/tests.yml:119]: Are you obsessed with 🏎️? 😄

**Review by ZhongRuoyu — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -88,6 +88,41 @@ jobs:
...
+      - name: Publish Analytics to Summary
+        env:
+          TESTING_FORMULAE: ${{needs.formulae_detect.outputs.testing_formulae}}
+        run: |
+          if [[ -z "$TESTING_FORMULAE" ]]
+          then
+              exit 0
+          fi
+
+          {
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:104]: I think this could probably all be replaced with `brew info --analytics $TESTING_FORMULAE`?

Comment (SMillerDev) [reply .github/workflows/tests.yml:104]: I tried, but that immediately puts it in tables that aren't as much more verbose than this one. And `brew info --json` doesn't seem to have that data.

**Pull request merged**